sor-api/src/main/java/com/bazaarvoice/emodb/sor/condition/impl/ComparisonConditionImpl.java

```diff
@@ -52,6 +52,92 @@ public void appendTo(Appendable buf)
     }
 
     @Override
+    public boolean overlaps(ComparisonCondition condition) {
+        Comparison otherComparison = condition.getComparison();
+        Object otherValue = condition.getValue();
+
+        if (_value instanceof Number ^ otherValue instanceof Number) {
+            // The other condition is for a different data type, so they can't overlap
+            return false;
+        }
+
+        // If both have the same direction then logically they must overlap at some point.
+        boolean isGreater = _comparison == Comparison.GT || _comparison == Comparison.GE;
+        boolean isOtherGreater = otherComparison == Comparison.GT || otherComparison == Comparison.GE;
+
+        if (isGreater == isOtherGreater) {
+            return true;
+        }
+
+        //noinspection unchecked
+        int compare = isGreater ? ((Comparable) otherValue).compareTo(_value) : ((Comparable) _value).compareTo(otherValue);
+
+        return compare > 0 ||
+                (compare == 0 && _comparison.isClosed() && otherComparison.isClosed());
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public boolean isSubsetOf(ComparisonCondition condition) {
+        Object lObject = getValue();
+        Object rObject = condition.getValue();
+
+        Comparable lValue;
+        Comparable rValue;
+
+        if (lObject instanceof Number) {
+            if (!(rObject instanceof Number)) {
```

Comment:
I though its easy to read the code that way with only 2 if cases and one else for anything else.